TfsMigrate.Processors.WorkItemTracking/WorkItemTrackingProcessor.cs

```diff
@@ -440,69 +449,107 @@ private Task LoadUsersAsync ( MigrationContext context, IEnumerable<UserSettings
                 var isChild = relation.IsChild();
                 var isParent = !isChild && relation.IsParent();
                 var isRelated = !isChild && !isParent && relation.IsRelated();
+                var isAttachment = relation.IsAttachment();
 
                 //Not a relation we care about                
-                if (!isChild && !isParent && !isRelated)
+                if (!isChild && !isParent && !isRelated && !isAttachment)
                     continue;
 
-                //Get the related item information
-                var relatedId = relation.GetRelatedId();
-                if (relatedId <= 0)
-                    continue;
-
-                var related = context.GetMigratedWorkItem(relatedId);
-                
-                if (isChild)
+                if (isAttachment)
                 {
-                    //If the child has been migrated then add a child link to the related item
-                    //else if the target isn't closed (or the override setting is set) then add the related item to migration list
-                    if (related != null)
-                    {
-                        if (related.TargetId != 0)
-                        {
-                            Logger.Debug($"Adding Child link to {related.TargetId}");
-                            doc.AddLink(WorkItemRelations.Child, related.TargetUrl);
-                        } else
-                            Logger.Warning($"Skipping Child link to {related.SourceId} because it failed to migrate");
-                    } else if (!targetItem.IsClosed() || Settings.IncludeChildLinksOnClosed)
+                    var guid = relation.GetRelatedGuid();
+                    var fileName = relation.Attributes["name"].ToString();
+                    var attributs = relation.Attributes.Where(x => x.Key != "id").ToDictionary(x => x.Key, x => x.Value);
+
+                    if (guid == Guid.Empty)
                     {
-                        Logger.Debug($"Adding Child link {relatedId} to migration list");
-                        AddToMigrationQueue(context.MigrationQueue, new MigratedWorkItem() { SourceId = relatedId });
-                    };
-                } else if (isParent)
+                        continue;
+                    }
+
+                    var readOnlyAttachmentFile = await context.SourceService.TryCatchAsync(c => c.GetAttachmentContentAsync(guid, cancellationToken: cancellationToken)).ConfigureAwait(false);
+
+                    MemoryStream attachmentFile = new MemoryStream();

```

Comment:
This is leaking a stream. Use a using statement to ensure the attachment memory is cleaned up after the attachment is added to the work item.